src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1862,32 +1862,102 @@ export default class AircraftModel {
      * @return {number}
      */
     _calculateTargetedAltitudeVnav() {
-        const nextAltitudeMaximumWaypoint = this.fms.findNextWaypointWithMaximumAltitudeAtOrBelow(this.altitude);
-        const nextAltitudeMinimumWaypoint = this.fms.findNextWaypointWithMinimumAltitudeAtOrAbove(this.altitude);
+        const nextAltitudeMaximumWaypoint = this.fms.findNextWaypointWithMaximumAltitudeRestriction();
+        const nextAltitudeMinimumWaypoint = this.fms.findNextWaypointWithMinimumAltitudeRestriction();
         const maximumAltitudeExists = !_isNil(nextAltitudeMaximumWaypoint);
         const minimumAltitudeExists = !_isNil(nextAltitudeMinimumWaypoint);
 
-        if (!maximumAltitudeExists && !minimumAltitudeExists) {
-            return this.mcp.altitude;
-        }
+        if (this.mcp.altitude < this.altitude) {
```

Comment:
holy nested `if` logic!  I'd *_really_* like to see this broken up into several other methods, all with as little conditional logic as possible.

Reply:
The branches are all needed, and it's broken up really cleanly and commented well. I think it likely makes the most _sense_ in terms of readding and figuring out what the heck is going on as it is here. That said, we typically would avoid nested ifs like these as if they were the plague.

My two cents is that we roll with it for now.